pkg/controller/windowsmachineconfig/nodeconfig/network.go

```diff
@@ -0,0 +1,137 @@
+	if hostSubnet == "" || validateCIDR(hostSubnet) != nil {
+		return errors.Errorf("error receiving valid value for node hostSubnet")
+	}
+	nw.hostSubnet = hostSubnet
+	return nil
+}
+
+// validateCIDR uses the parseCIDR from network package
+// to validate the format of the CIDR received
+func validateCIDR(serviceCIDR string) error {
+	_, _, err := net.ParseCIDR(serviceCIDR)
+	if err != nil || serviceCIDR == "" {
+		return errors.Wrapf(err, "error receiving valid CIDR value")
+	}
+	return nil
+}
+
+// populateCniConfig populates the CNI config template with necessary information and
+// creates a new file in temp directory to store the modified template
+func (nw *network) populateCniConfig(serviceCIDR string, templatePath string) (string, error) {
+	if nw.hostSubnet == "" {
+		return "", errors.New("received empty value for node hostSubnet, can't " +
+			"populate CNI config template")
+	}
+
+	if validateCIDR(serviceCIDR) != nil || serviceCIDR == "" {
+		return "", errors.New("received invalid value for cluster service CIDR, can't " +
+			"populate CNI config template")
+	}
+
+	cniConfTemplate, err := os.Open(templatePath)
+	if err != nil {
+		return "", errors.Wrapf(err, "error opening CNI config template from %s", templatePath)
+	}
+	defer cniConfTemplate.Close()
+	decoder := json.NewDecoder(cniConfTemplate)
+	cniCfg := cniConf{}
+
+	if err = decoder.Decode(&cniCfg); err != nil {
+		return "", errors.Wrap(err, "can't decode cniConf JSON struct")
+	}
+
+	if len(cniCfg.Policies) < 2 || len(cniCfg.Policies[0].Value.ExceptionList) == 0 ||
+		cniCfg.Policies[1].Value.DestinationPrefix == "" {
+		return "", errors.Errorf("invalid policy fields in cniConf struct" +
+			" to modify CNI config template")
+	}
+
+	cniCfg.IPAM.Subnet = nw.hostSubnet
+	cniCfg.Policies[0].Value.ExceptionList[0] = serviceCIDR
+	cniCfg.Policies[1].Value.DestinationPrefix = serviceCIDR
+
+	// retrieve the json file from the modified struct
+	response, err := json.Marshal(&cniCfg)
+	if err != nil {
+		return "", errors.Wrap(err, "can't retrieve cniConf JSON using modified struct")
+	}
+
+	// Create a temp file to hold the cniCfg
+	tmpCniDir, err := ioutil.TempDir("", "cni")
+	if err != nil {
+		return "", errors.Wrapf(err, "error creating Local temp CNI directory")
+	}
+	cniConfigPath, err := os.Create(filepath.Join(tmpCniDir, "cni-conf-template.json"))
+	if err != nil {
+		return "", errors.Wrapf(err, "error creating Local cni-conf-template.json file")
+	}
+	if _, err = cniConfigPath.Write(response); err != nil {
+		return "", errors.Wrapf(err, "can't write JSON CNI config file to %s", cniConfigPath.Name())
+	}
+	if err := cniConfigPath.Close(); err != nil {
+		return "", errors.Wrapf(err, "can't close CNI config file %s", cniConfigPath.Name())
+	}
+	if cniConfigPath.Name() != "" {
+		return cniConfigPath.Name(), nil
+	}
+	return "", errors.Errorf("couldn't retrieve modified CNI config file %s", cniConfigPath.Name())
```

Comment:
Please swap. It is good practice to  check and return for errors first. I am also not sure if the error message is correct. Isn't the issue that you got an empty filename?